### DISCUSSION TITLE
Actual memory used when refreshing a year partition of 10 partitioned years

### DISCUSSION THREAD

**Discussion by TinyTimothy2** [TabularEditor/TabularEditor3#1394] [Data Modeling]
In SSAS where you have partitions, say 10 years, if you are refreshing full only 1 partition does it use less memory than all 10?  Ie it only pins the refreshing current partition in memory before it swaps the refreshing partition?
I don't have a tool to see what is actually happening.
If 10 years is 100 GB and each year is 10 GB what is the breakdown of memory and how that effects users using the cube.
And wouldn't it affect all users of all cubes as all the memory in total is 'shared' across SSAS settings?

**Comment by otykier** [comment]
The rule of thumb for SSAS memory sizing, is to have a total memory of 2,5x the size of the model at rest. (I.e. if you have a model which is 10 GB total size, then 25 GB should be enough to ensure you can perform a full refresh of the entire model, while still leaving 5 GB to serve queries that are coming in from users). But this assumes that you have a machine that's dedicated to SSAS (i.e. it doesn't run other services like SQL Server relational engine, SSRS, etc.)

Of course, this rule is impractical if you have large fact tables of 100+ GB. And that's where partitioning comes in.

> In SSAS where you have partitions, say 10 years, if you are refreshing full only 1 partition does it use less memory than all 10? Ie it only pins the refreshing current partition in memory before it swaps the refreshing partition?

Correct.

In theory, if you're only refreshing one partition of 10 GB, on a 100 GB model, the memory used would be at most ~110 GB.

[Here is a detailed article on SSAS memory settings](https://www.sqlbi.com/articles/optimizing-memory-settings-in-analysis-services/).

**Comment by TinyTimothy2** [comment]
After a couple of years I can definitively say that I still do not have definitive answer :  )
I am fairly certain that even refreshing 1 partition, or 1 small date table or 1 measure pins the whole model object in memory until it swaps.
Small table and measures happen so fast that you don't notice.
